internal/store/ingress.go

```diff
@@ -202,6 +202,28 @@ func ingressMetricFamilies(allowAnnotationsList, allowLabelsList []string) []gen
 				}
 			}),
 		),
+		*generator.NewFamilyGeneratorWithStability(
+			"kube_ingress_status",
+			"Ingress status.",
+			metric.Gauge,
+			basemetrics.STABLE,
+			"",
+			wrapIngressFunc(func(i *networkingv1.Ingress) *metric.Family {
+				ms := []*metric.Metric{}
+				for _, ingress := range i.Status.LoadBalancer.Ingress {
+					for _, port := range ingress.Ports {
+						ms = append(ms, &metric.Metric{
+							LabelKeys:   []string{"ip", "hostname", "port", "protocol"},
```

Comment:
Do you need both the `ip` & the `hostname` ?

Reply:
Wondering since IPs can keep on changing so cardinality might explode. Just using hostname might suffice?

Reply:
At first, I just wanted to include all the fields, but as you mentioned, the hostname will suffice.
Made the changes here: https://github.com/kubernetes/kube-state-metrics/pull/2433/commits/622505e6ce1e0df535e61d69e636a5962c70ce50

Reply:
If we're removing IPs could we end up with duplicate metrics if multiple LB VIPs serve the same hostname (not sure if this is likely at all, maybe with multiple regionally deployed LBs)?

Reply:
I don't think we should include IPs (due to the cardinality concern that Richa pointed out).

> If we're removing IPs could we end up with duplicate metrics if multiple LB VIPs serve the same hostname (not sure if this is likely at all, maybe with multiple regionally deployed LBs)?

Won't we have unique instance IDs?

Reply:
With instance IDs you mean in this object https://pkg.go.dev/k8s.io/api/networking/v1#Ingress ?

I don't know if that is the case, I assume there's must be some reason why the object is so convoluted...

https://pkg.go.dev/k8s.io/api/networking/v1#IngressLoadBalancerStatus contains a list of https://pkg.go.dev/k8s.io/api/networking/v1#IngressLoadBalancerIngress (both IP and Hostname are optional, without having looked into the code, I assume you need to set either one or the other?)
which then contains a list of https://pkg.go.dev/k8s.io/api/networking/v1#IngressPortStatus

Reply:
I think we should keep both `ip` and `hostname` otherwise this metric won't make much sense since we won't be able to distinguish between the load balancers.
If cardinality is a concern we can always have this metric disabled by default, but here I don't think it is a problem because the number of timeseries for this metric will be limited to the number of Endpoints in the cluster. This is just the theoretical worst case scenario, but in practice it will only be a subset of that since it only includes the endpoints of the loadbalancer. and since the number of Endpoints in a cluster is limited due to scalability limits, this metric will be bounded.

Disclaimer: I am not too knowledgeable about Ingress so for what I know it might just be bounded to the number of Services instead of Endpoints which is even better. It all depends on where the <ip/hostname>:<port> pair come from.